accounting/src/creditcommons/credit-commons-controller.ts

```diff
@@ -61,45 +126,69 @@ export class CreditCommonsControllerImpl extends AbstractCurrencyController impl
     await this.checkLastHashAuth(ctx)
     return { message: 'Welcome to the Credit Commons federation protocol.' }
   }
-  async getAccountHistory(ctx: Context) {
+  async getAccountHistory(ctx: Context, accountId: string) {
     await this.checkLastHashAuth(ctx)
+    const { transfersIn, transfersOut } = await this.getTransactions(accountId)
+    const transfers = transfersIn.concat(transfersOut.map(t => { t.amount = -t.amount; return t }))
+    let data: {
+      [created: string]: number
+    } = {}
+    let min = 0
+    let max = 0
+    let start = new Date('9999')
+    let end = new Date('0000')
+
+    transfers.forEach((t: Transfer) => {
+      const amount = parseFloat(this.currencyController.amountToLedger(t.amount))
+      data[formatDateTime(t.created)] = amount
+      if (amount < min) { min = amount }
+      if (amount > max) { max = amount }
+      if (t.created < start) { start = t.created }
+      if (t.created > end) { end = t.created }
+    })
     return {
-      data: {
-        '2025-04-01 09:22:37': 0,
-        '2025-04-01 09:22:41': 0,
-        '2025-04-01 09:22:42': -60
-      },
+      data,
       meta: {
-        min: -60,
-        max: 0,
+        min,
+        max,
         points: 3,
-        start: '2025-04-01 09:22:37',
-        end: '2025-04-01 09:22:42'
+        start: formatDateTime(start),
+        end: formatDateTime(end)
       }
     };
   }
+  async codeToAccountId(code: string): Promise<string | undefined> {
+    const record: AccountRecord | null = await this.db().account.findUnique({
+      where: { 
+        code,
+        status: "active",
+      },
+      include: {
+      }
+    })
+    return record?.id
+  }
   async createTransaction(ctx: Context, transaction: CreditCommonsTransaction) {
     this.gatewayAccountId = await this.checkLastHashAuth(ctx)
-    
-    let localTransfers: InputTransfer[] = []
-    let netGain = 0;
-    let recipient = null;
+    let netGain = 0
+    let recipient = null
+    let metas: string[] = []
+    let froms: string[] = []
     for (let i=0; i < transaction.entries.length; i++) {
       let payer, payee, thisRecipient;
       if (transaction.entries[i].payer.startsWith(this.ledgerBase)) {
-        payer = transaction.entries[i].payer.slice(this.ledgerBase.length)
-        payee = this.gatewayAccountId
-        thisRecipient = payer
+        thisRecipient = transaction.entries[i].payer.slice(this.ledgerBase.length)
         netGain -= transaction.entries[i].quant
+        metas.push(`-${transaction.entries[i].quant} (${transaction.entries[i].description})`)
       }
       if (transaction.entries[i].payee.startsWith(this.ledgerBase)) {
-        payee = transaction.entries[i].payee.slice(this.ledgerBase.length)
-        payer = this.gatewayAccountId
         if (thisRecipient) {
           throw new Error('Payer and Payee cannot both be local')
```

Comment:
```
throw badRequest('Payer and Payee cannot both be local') // from error.ts
```

Reply:
this way it assigns the good error code and title `400 Bad request` in this case